src/platform/mbed/SystemTimeSupport.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides implementations of the CHIP System Layer platform
+ *          time/clock functions that are suitable for use on the Mbed-OS platform.
+ */
+/* this file behaves like a config.h, comes first */
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include <support/TimeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+#include <Kernel.h>
+
+namespace chip {
+namespace System {
+namespace Platform {
+namespace Layer {
+
+uint64_t GetClock_Monotonic()
+{
+    return rtos::Kernel::get_ms_count() * UINT64_C(1000);
+}
+
+uint64_t GetClock_MonotonicMS()
+{
+    return rtos::Kernel::get_ms_count();
+}
+
+uint64_t GetClock_MonotonicHiRes()
+{
+    return GetClock_Monotonic();
+}
+
+Error GetClock_RealTime(uint64_t & curTime)
+{
+    struct timeval tv;
+    int res = gettimeofday(&tv, NULL);
+    if (res != 0)
+    {
+        return CHIP_SYSTEM_ERROR_UNEXPECTED_STATE;
+    }
+    if (tv.tv_sec < CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD)
+    {
+        return CHIP_SYSTEM_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    curTime = (tv.tv_sec * UINT64_C(1000000)) + tv.tv_usec;
+    return CHIP_SYSTEM_NO_ERROR;
+}
+
+Error GetClock_RealTimeMS(uint64_t & curTimeMS)
+{
+    struct timeval tv;
+    int res = gettimeofday(&tv, NULL);
+    if (res != 0)
+    {
+        return CHIP_SYSTEM_ERROR_UNEXPECTED_STATE;
+    }
+    if (tv.tv_sec < CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD)
+    {
+        return CHIP_SYSTEM_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    curTimeMS = (tv.tv_sec * UINT64_C(1000)) + (tv.tv_usec / 1000);
```

Comment:
Indeed, UINT64_C() adds ULL suffix.
I think we don't need UINT64_C() on second constant  because there is no overflow risk in that case (even if type_t may be resolved in different basic type, depending on platform.